apiserver/facades/agent/caasoperator/operator_test.go

```diff
@@ -290,3 +298,25 @@ func (s *CAASOperatorSuite) TestWatch(c *gc.C) {
 	resource := s.resources.Get("1")
 	c.Assert(resource, gc.Equals, s.st.app.watcher)
 }
+
+func (s *CAASOperatorSuite) TestSetTools(c *gc.C) {
+	vers := version.Binary{}
+	results, err := s.facade.SetTools(params.EntitiesVersion{
+		AgentTools: []params.EntityVersion{
+			{Tag: "application-gitlab", Tools: &params.Version{}},
```

Comment:
Setting to a zero version seems like a problem to me.

Reply:
fixed